compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1207,6 +1207,18 @@ impl<'tcx> ToPredicate<'tcx, PolyTraitPredicate<'tcx>> for Binder<'tcx, TraitRef
     }
 }
 
+impl<'tcx> ToPredicate<'tcx, PolyTraitPredicate<'tcx>> for TraitRef<'tcx> {
+    fn to_predicate(self, tcx: TyCtxt<'tcx>) -> PolyTraitPredicate<'tcx> {
+        ty::Binder::dummy(self).to_predicate(tcx)
```

Comment:
Nope, this is not a change we should merge (or the impl below).

In fact, I would be in favor of some explicit negative impl here just so we can slap a big comment on this so nobody tries to make the positive impl again.

Reply:
Can you explain what is wrong in this impl?

Reply:
Ah, I see: https://github.com/rust-lang/rust/pull/110806#discussion_r1177875507